### PR TITLE
Add InjectableGenerator

### PR DESCRIPTION
**Pull request by miguelrochefort**

### CODE REVIEW

File: src/Uno.Injectable/IInjectable.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	public interface IInjectable
+    {
+		/// <summary>
+		/// Inject the dependency resolver used to populate properties and fields attributed with <see cref="Uno.InjectAttribute"/>.
+		/// </summary>
+		/// <remarks>
+		/// You are responsible for calling this method and provide the appropriate dependency resolver.
+		/// </remarks>
+		/// <param name="resolver">The dependency resolver</param>
+		void Inject(Func<Type, string, object> resolver);
```
Comment (dr1rrb) [inline src/Uno.Injectable/IInjectable.cs]: Should we declare a dedicated delegate for that ?

Comment (carldebilly) [reply src/Uno.Injectable/IInjectable.cs]: I'm for it, too.

File: src/Uno.Injectable/IInjectable.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+namespace Uno
+{
+	/// <summary>
+	/// Exposes a way to inject a dependency resolver.
+	/// </summary>
+	/// <remarks>
+	/// A partial class implementing <see cref="Uno.IInjectable"/> is generated for every class defining properties or fields attributed with <see cref="Uno.InjectAttribute"/>.
+	/// </remarks>
+	public interface IInjectable
+    {
```
Comment (dr1rrb) [inline src/Uno.Injectable/IInjectable.cs]: Alignment issue :)

File: src/Uno.Injectable/IInjectable.cs (hunk 1 above)
Comment (miguelrochefort) [reply src/Uno.Injectable/IInjectable.cs]: I added the `Uno.DependencyResolver` delegate.

File: src/Uno.Injectable/IInjectable.cs (hunk 2 above)
Comment (miguelrochefort) [reply src/Uno.Injectable/IInjectable.cs]: Fixed.

File: src/Uno.Injectable/InjectAttribute.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+namespace Uno
+{
+	/// <summary>
+	/// Indicates that the attributed property or field should be injected.
+	/// </summary>
+	/// <remarks>
+	/// For partial class implementing <see cref="Uno.IInjectable" /> is generated for every class that defines properties or fields attributed with <see cref="Uno.InjectAttribute"/>.
+	/// </remarks>
+	[AttributeUsage(AttributeTargets.Field | AttributeTargets.Property, AllowMultiple = false)]
+	public sealed class InjectAttribute : Attribute
```
[the remark this answers is not in the document]
Comment (miguelrochefort) [reply src/Uno.Injectable/InjectAttribute.cs]: [Inject] is the attribute used by the popular Dagger 2.0 framework.

[Dagger 2.0](https://github.com/google/dagger)
[Stiletto (C# port of Dagger)](https://github.com/benjamin-bader/stiletto)

File: src/Uno.Injectable/Uno.Injectable.csproj [hunk 4]
```diff
@@ -0,0 +1,25 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+	<PropertyGroup>
+		<TargetFrameworks>net46;netstandard1.3;netstandard2.0</TargetFrameworks>
```
[the remark this answers is not in the document]
Comment (miguelrochefort) [reply src/Uno.Injectable/Uno.Injectable.csproj]: C'est les memes targets que pour Uno.Immutable et Uno.Equality.

**Pull request merged**